index.js

```diff
@@ -13,40 +13,41 @@ const factory = exports.factory = (struct, key) => {
     };
 };
 
```

Comment:
Moved this just above where it's used. Also changed the order of the params to reflect when one would have the data in question (you'd likely have `encryptionKey` and `iv` well before before you had `buffer`) IMHO ordering them this way makes it easier for callers to do partial application as well.